vpc-endpoints.tf

```diff
@@ -14,8 +14,12 @@ resource "aws_vpc_endpoint" "s3" {
   vpc_id            = local.vpc_id
   service_name      = data.aws_vpc_endpoint_service.s3[0].service_name
   vpc_endpoint_type = var.s3_endpoint_type
-
-  tags = local.vpce_tags
+  tags = merge(
+    {
+      "Name" = format("%s-%s", var.name, data.aws_vpc_endpoint_service.s3[0].service)
+    },
+    local.vpce_tags
```

Comment:
so this does work but theres one slight issue - if you try to test this using the `examples/complete-vpc` you won't see your changes (all endpoints will still be named `complete`). we have a deeply nested overloading of tags going on currently (`var.tags` -> `var.vpc_endpoint_tags` -> `Name` set here). I think to fix and decouple this change a little better, lets do this:

```hcl
  tags = var.vpce_name_include_service ? merge(
    local.vpce_tags,
    {
      "Name" = format("%s-%s", var.name, data.aws_vpc_endpoint_service.codedeploy[0].service)
    },
  ) : local.vpce_tags
```

And we'll need to add the variable block for the `vpce_name_include_service` variable where the default is `false`. This means that today if users don't want to change their VPC endpoint names (aka - leave them as they have been up till now), then this change is transparent to them. however, if users want to opt into this change and have the endpoint service name appended to the VPC endpoint name, they can flip the switch by setting `vpce_name_include_service = true`

Reply:
Yep, good idea. I'll get a new PR put up for this. Thanks